### DISCUSSION TITLE
Any Way to Detect Inject/Clipboard Text Replacement Through Windows Event Viewer or Some Other Service?

### DISCUSSION THREAD

**Discussion by andrewcincotta** [espanso/espanso#1469] [Q&A] [answered]
I am trying to add a text expansion sound to Espanso, kind of like aText's or Alfred's. I am not the best with Rust, so I haven't really tried to compile my own version of Espanso with sound effects, but I do have an EXE that will play the sound effect audio file I want. With that I am trying to find a way to run scripts/programs after a text expansion from espanso, so that I can run that EXE and play the sound effect. Maybe there's a way with Windows Event Viewer?

**Comment by smeech** [comment, marked as answer]
Sorry about the late reply - I am trying to mop up and sort the old Issues and Discussions.

Thank you. There have been several requests for an option to make a sound when expansions are triggered.

For reference: #221, #1465, #1170.
